openeo/UDP.ipynb

```diff
@@ -0,0 +1,352 @@
+       "        font.as = \"font\";\n",
+       "        font.type = \"font/woff2\";\n",
+       "        font.crossOrigin = true;\n",
+       "        font.href = \"https://use.fontawesome.com/releases/v5.13.0/webfonts/fa-solid-900.woff2\"\n",
+       "        document.head.appendChild(font);\n",
+       "    }\n",
+       "    </script>\n",
+       "    <openeo-process>\n",
+       "        <script type=\"application/json\">{\"show-graph\": true, \"provide-download\": false, \"process\": {\"id\": \"NDWI\", \"links\": [{\"href\": \"https://openeo-staging.creo.vito.be/openeo/1.1/processes/u:3e24e251-2e9a-438f-90a9-d4500e576574/NDWI\", \"rel\": \"canonical\", \"title\": \"Public URL for user-defined process 'NDWI'\"}], \"parameters\": [{\"description\": \"A date within wich you wish to load the Terrasope NDVI product.\", \"name\": \"date\", \"schema\": {\"type\": \"string\"}}, {\"description\": \"Spatial extent for area of interst to calculate ndvi\", \"name\": \"aoi\", \"schema\": {\"subtype\": \"geojson\", \"type\": \"object\"}}], \"process_graph\": {\"loadcollection1\": {\"arguments\": {\"bands\": [\"B03\", \"B08\"], \"id\": \"SENTINEL2_L2A\", \"spatial_extent\": {\"from_parameter\": \"aoi\"}, \"temporal_extent\": {\"from_parameter\": \"date\"}}, \"process_id\": \"load_collection\"}, \"reducedimension1\": {\"arguments\": {\"data\": {\"from_node\": \"loadcollection1\"}, \"dimension\": \"t\", \"reducer\": {\"process_graph\": {\"max1\": {\"arguments\": {\"data\": {\"from_parameter\": \"data\"}}, \"process_id\": \"max\", \"result\": true}}}}, \"process_id\": \"reduce_dimension\"}, \"reducedimension2\": {\"arguments\": {\"data\": {\"from_node\": \"reducedimension1\"}, \"dimension\": \"bands\", \"reducer\": {\"process_graph\": {\"add1\": {\"arguments\": {\"x\": {\"from_node\": \"arrayelement1\"}, \"y\": {\"from_node\": \"arrayelement2\"}}, \"process_id\": \"add\"}, \"arrayelement1\": {\"arguments\": {\"data\": {\"from_parameter\": \"data\"}, \"index\": 0}, \"process_id\": \"array_element\"}, \"arrayelement2\": {\"arguments\": {\"data\": {\"from_parameter\": \"data\"}, \"index\": 1}, \"process_id\": \"array_element\"}, \"divide1\": {\"arguments\": {\"x\": {\"from_node\": \"subtract1\"}, \"y\": {\"from_node\": \"add1\"}}, \"process_id\": \"divide\", \"result\": true}, \"subtract1\": {\"arguments\": {\"x\": {\"from_node\": \"arrayelement1\"}, \"y\": {\"from_node\": \"arrayelement2\"}}, \"process_id\": \"subtract\"}}}}, \"process_id\": \"reduce_dimension\", \"result\": true}}, \"public\": true}}</script>\n",
+       "    </openeo-process>\n",
+       "    "
+      ],
+      "text/plain": [
+       "<openeo.rest.udp.RESTUserDefinedProcess at 0x2d19831ef70>"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# publishing the service\n",
+    "process_name = \"NDWI\"\n",
+    "my_graph = ndwi.flat_graph()\n",
```

Comment:
`my_graph = ndwi.flat_graph()`

this is not necessary, you can directly pass `ndwi` to `save_user_defined_process`